crates/pallet-domains/src/lib.rs

```diff
@@ -369,7 +369,7 @@ impl<T: Config> Pallet<T> {
     ) -> bool {
         receipts
             .array_windows()
-            .all(|[ref head, ref tail]| head.primary_number + One::one() == tail.primary_number)
+            .all(|[ref head, ref tail]| head.primary_number + One::one() == tail.primary_number) // saturating_add
```

Comment:
`saturating_add` is a bad default (that I see you suggested everywhere), in most cases we'd rather fail than silently ignore the error. However, we should indeed use `checked_add` and `.expect()` in such cases, explaining why overflow isn't possible. Thanks!

I didn't check, but maybe clippy has a lint to not allow unchecked math at all, if so we'll enable it going forward to ensure we do not miss it accidentally anywhere.

Reply:
You are right, probably in most cases `checked_add` would be better. I just copied that comment to mark unsafe math operations. It would be great if clippy caught such cases, but I don't know anything about a linter for this.